integrations/integration.php

```diff
@@ -135,18 +156,114 @@ public function get_slug(): string {
 	}
 
 	/**
-	 * Set vip_config property.
+	 * Set `vip_configs` property.
 	 *
-	 * @param IntegrationVipConfig $vip_config Instance of IntegrationVipConfig.
+	 * @param array<VipIntegrationConfig> $vip_configs Configurations provided by VIP.
 	 *
 	 * @return void
+	 *
+	 * @private
 	 */
-	public function set_vip_config( IntegrationVipConfig $vip_config ): void {
-		if ( ! $this->is_active() ) {
-			trigger_error( sprintf( 'Configuration info can only assigned if integration is active.' ), E_USER_WARNING ); // phpcs:ignore WordPress.PHP.DevelopmentFunctions.error_log_trigger_error
+	public function set_vip_configs( array $vip_configs ): void {
+		$this->vip_configs = $vip_configs;
+	}
+
+	/**
+	 * Get statuses of the integration in context of current site.
+	 *
+	 * @return array<Env_Integration_Status>
+	 */
+	private function get_site_statuses() {
+		/**
+		 * Statuses of the integration.
+		 *
+		 * @var array<string>
+		 */
+		$statuses = [];
+
+		foreach ( $this->vip_configs as $vip_config ) {
+			if ( $this->get_value_from_config( $vip_config, 'org', 'status' ) === Org_Integration_Status::BLOCKED ) {
+				return [ Org_Integration_Status::BLOCKED ];
+			}
+	
+			// Look into network_sites config before and then fallback to env config.
+			$statuses[] = $this->get_value_from_config( $vip_config, 'network_sites', 'status' ) ??
+				$this->get_value_from_config( $vip_config, 'env', 'status' );
+		}
+
+		return $statuses;
+	}
+
+	/**
+	 * Get configs of the integration in context of current site.
+	 *
+	 * @return array<VipIntegrationConfig> Returns an array of configs if the integration have multiple configs else single config object.
+	 *
+	 * @private
+	 */
+	public function get_site_configs() {
+		/**
+		 * Array containing configs of the integration.
+		 *
+		 * @var array<VipIntegrationConfig>
+		 */
+		$configs = [];
+
+		// Get configs of the integration from configurations provided by VIP.
+		foreach ( $this->vip_configs as $vip_config ) {
+			if ( is_multisite() ) {
+				$config = $this->get_value_from_config( $vip_config, 'network_sites', 'config' );
+
+				// If network site config is not found then fallback to env config if it exists.
+				if ( empty( $config ) && true === $this->get_value_from_config( $vip_config, 'env', 'cascade_config' ) ) {
+					$config = $this->get_value_from_config( $vip_config, 'env', 'config' );
+				}
+			} else {
+				$config = $this->get_value_from_config( $vip_config, 'env', 'config' );
+			}
+
+			if ( isset( $config ) ) {
+				$configs[] = $config;
+			}
+		}
+
+		// Return config object if integration have only one config else return all configs.
+		return ( ! $this->have_multiple_setups && isset( $configs[0] ) ) ? $configs[0] : $configs;
```

Comment:
Besides Composable, how many other integrations will have multiple set-ups? Just wondering, but is this something that can be handled elsewhere? :/

Reply:
There are some nice benefits to having individual sources be able to cascade down with their own configs like the other integrations. Helps prevent duplication for network sites for example, and keeps things logically operating the same way.